File: core0/us/test/symmetric_encryption_dir/main.cpp
//===-                           S C R I P T  T.V.
//===-                           https://script.tv
//===-
//===-            Copyright (C) 2017-2024 Script Network
//===-            Copyright (C) 2017-2024 manicberet@gmail.com
//===-
//===-                      GNU GENERAL PUBLIC LICENSE
//===-                       Version 3, 29 June 2007
//===-
//===-    This program is free software: you can redistribute it and/or modify
//===-    it under the terms of the GPLv3 License as published by the Free
//===-    Software Foundation.
//===-
//===-    This program is distributed in the hope that it will be useful,
//===-    but WITHOUT ANY WARRANTY; without even the implied warranty of
//===-    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
//===-
//===-    You should have received a copy of the General Public License
//===-    along with this program, see LICENCE file.
//===-    see https://www.gnu.org/licenses
//===-
//===----------------------------------------------------------------------------
//===-
#include "us/gov/crypto/symmetric_encryption.h"

#include <iostream>
#include <string>
#include "us/gov/crypto/ec.h"
#include "us/gov/crypto/base58.h"
#include <vector>
#include <array>

#define loglevel "test"
#define logclass "symm_enc_main"
#include <us/gov/logs.inc>


using namespace std;
using namespace us::gov::crypto;

//using namespace us::test;

using namespace std;
using namespace us::gov::crypto;

using us::ko;
using us::is_ko;
using us::is_ok;

int main ( int argc, char *argv[] )
{

    us::dbg::thread_logger::set_root_logdir("logs");
    log_pstart("generate_keys_main");
    log_start("", "main");

    if ( argc < 5 ) {
        cout<<"required arugments: <private key> <public key> <[encrypt|decrypt]> <message>" << endl;
        if(argc == 2)
            cout<<"provided private key: " << argv[1] << endl;
        if(argc == 3)
            cout<<"provided private key: " << argv[1] << "\n" << "provided public key: " << argv[2] << endl;
        if(argc == 4)
            cout<<"provided private key: " << argv[1] << "\n" << "provided public key: " << argv[2] << "\n" << "provided command: " << argv[3] << endl;
    }
    else {

        ec::keys k;
        k.priv=ec::keys::priv_t::from_b58(argv[1]);
        k.pub=ec::keys::pub_t::from_b58(argv[2]);

        string command(argv[3]);

        symmetric_encryption s_e;
        s_e.init(k.priv, k.pub);

            if (command == "encrypt") {
                string message_string(argv[4]);
                vector<unsigned char> message(message_string.begin(),message_string.end());
                vector<unsigned char> encrypted;
                ko r = s_e.encrypt(message, encrypted, 0);
                assert(is_ok(r));
                cout << b58::encode(encrypted) << endl;
            }
            if (command == "decrypt") {
                vector<unsigned char> message;
                b58::decode(argv[4], message);
                vector<unsigned char> decrypted;
                ko r = s_e.decrypt(message, decrypted);
                assert(is_ok(r));
                string decrypted_string(decrypted.begin(), decrypted.end());
                cout << decrypted_string << endl;
            }

    }
}

File: core0/us/test/symmetric_encryption_dir/Main.java
//===-                           S C R I P T  T.V.
//===-                           https://script.tv
//===-
//===-            Copyright (C) 2017-2024 Script Network
//===-            Copyright (C) 2017-2024 manicberet@gmail.com
//===-
//===-                      GNU GENERAL PUBLIC LICENSE
//===-                       Version 3, 29 June 2007
//===-
//===-    This program is free software: you can redistribute it and/or modify
//===-    it under the terms of the GPLv3 License as published by the Free
//===-    Software Foundation.
//===-
//===-    This program is distributed in the hope that it will be useful,
//===-    but WITHOUT ANY WARRANTY; without even the implied warranty of
//===-    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
//===-
//===-    You should have received a copy of the General Public License
//===-    along with this program, see LICENCE file.
//===-    see https://www.gnu.org/licenses
//===-
//===----------------------------------------------------------------------------
//===-
import us.wallet.*;
import us.gov.crypto.base58;
import us.gov.crypto.ec;
import us.gov.crypto.symmetric_encryption;
import java.security.PrivateKey;
import java.security.PublicKey;
import java.security.GeneralSecurityException;
import us.pair;
import us.ko;

public class Main{
    public static void main(String [ ] args) throws GeneralSecurityException{
        if (args.length <= 3) {
            return;
        }
                ec.create_instance();

                PrivateKey privateKey = ec.instance.get_private_key(args[0]);
                PublicKey publicKey = ec.instance.get_public_key(args[1]);


//                byte[] priv=base58.decode(args[0]);
//                byte[] pub=base58.decode(args[1]);
                String command = args[2];

//                PrivateKey privateKey = EllipticCryptography.secp256k1.getPrivateKey(priv);
//                PublicKey publicKey = EllipticCryptography.secp256k1.getPublicKey(pub);

                symmetric_encryption se = new symmetric_encryption();
                se.init(privateKey, publicKey);

                if (command.equals("decrypt")) {

                    byte[] message = base58.decode(args[3]);

                    pair<ko, byte[]> r = se.decrypt(message);
                    byte[] decrypted = r.second;
                    String decrypted_string = new String(decrypted);
                    System.out.println(decrypted_string);
                }
                if (command.equals("encrypt")) {

                    byte[] message = args[3].getBytes();

                    pair<ko, byte[]> r = se.encrypt(message);
                    byte[] encrypted = r.second;
                    System.out.println(base58.encode(encrypted));
                }



    }



}
